nevergrad/functions/ml/mlfunctionlib.py

```diff
@@ -174,16 +178,36 @@ def __init__(self, regressor: str, data_dimension: int):
             assert False, f"Problem type {regressor} undefined!"
         self.register_initialization(regressor=regressor, data_dimension=data_dimension)
 
-    def get_dataset(self, data_dimension):
+    def get_dataset(self, data_dimension, dataset_name):
+
+        if dataset_name != "artificial":
+            assert dataset_name in ["boston", "diabetes"]
+            data = {"boston": sklearn.datasets.load_boston,
+                    "diabetes": sklearn.datasets.load_diabetes,
+                    }[dataset_name](return_X_y=True)
+            # Half the dataset for training.
+            self.X = data[0][::2]
+            self.y = data[1][::2]
+            num_train_data = len(self.X)
+            self.num_data = num_train_data
+            for cv in range(10):
+                train_range = np.arange(num_train_data) % 10 != cv 
+                valid_range = np.arange(num_train_data) % 10 == cv
+                self.X_train += [self.X[train_range]]
+                self.y_train += [self.y[train_range]]
+                self.X_valid += [self.X[valid_range]]
+                self.y_valid += [self.y[valid_range]]
+            self.X_test = data[0][1::2]
+            self.y_test = data[1][1::2]
+            return
 
         num_data: int = 120  # Training set size.
         self.num_data = num_data
         
         # Training set.
         X = np.arange(0., 1., 1. / (num_data * data_dimension))
         X = X.reshape(-1, data_dimension)
-        random_state = np.random.RandomState(17)
-        random_state.shuffle(X)
+        np.random.shuffle(X)
```

Comment:
you need to use `self.parametrization.random_state` and avoid pulling from the default random state.
to do so, you may need to make sure you call `super` before `get_dataset` in the `init`

Reply:
I'm lost! In xpbase.py, I see:
==================
        if self.seed is not None and self._optimizer is None:
            # Note: when resuming a job (if optimizer is not None), seeding is pointless (reproducibility is lost)
            np.random.seed(self.seed)  # seeds both functions and parametrization (for which random state init is lazy)
            random.seed(self.seed)
            torch.manual_seed(self.seed)  # type: ignore
===========
which suggests that we are precisely pulling from the default random state.

Reply:
it's more a hack to make sure it works well in all cases, but it's much better to avoid it

Reply:
Ok I see. Fixed.